### PR TITLE
Add default support

### PR DESCRIPTION
**Pull request by dessen-xu**
**What this PR does / why we need it**:

Support default in instance name filter filed of datasource variable.

**Which issue(s) this PR fixes**:

https://github.com/grafana/grafana/issues/32623

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/variables/datasource/reducer.ts [hunk 1]
```diff
@@ -45,15 +45,19 @@ export const dataSourceVariableSlice = createSlice({
           continue;
         }
 
+        if (source.isDefault) {
+          if (regex === undefined) {
+            options.push({ text: 'default', value: 'default', selected: false });
+          } else if (regex && regex.exec('default')) {
+            options.push({ text: 'default', value: 'default', selected: false });
+          }
+        }
```
Comment (hugohaggmark) [inline public/app/features/variables/datasource/reducer.ts:54]: I would like some test coverage for:
- When a variable has `default` in regex and data source is default
- When a variable hasn't have `default` in regex and data source is default

**Review by hugohaggmark — COMMENTED** [review]
Looks good and works as expected, we just need to add some tests for this as well. Let me know if you need any help with that.

File: public/app/features/variables/datasource/reducer.ts (hunk 1 above)
Comment (dessen-xu) [reply public/app/features/variables/datasource/reducer.ts]: done

**Comment by dessen-xu** [comment]
done

**Comment by hugohaggmark** [comment]
Opened a small refactor PR for this in https://github.com/dessen-xu/grafana/pull/1

**Comment by dessen-xu** [comment]
done

**Comment by hugohaggmark** [comment]
Thank you for contributing @dessen-xu 🎉

**Comment by dessen-xu** [comment]
Thank you very much

**Review by hugohaggmark — APPROVED** [review]
LGTM!

**Review by thisisobate — APPROVED** [review]

**Pull request merged**